### DISCUSSION TITLE
Adafruit TinyUSB Library for Arduino

### DISCUSSION THREAD

**Discussion by rtek1000** [stm32duino/Arduino_Core_STM32#1912] [Q&A] [answered]
Does this core already have support for this library?

> Cores with built-in support
> Following core has TinyUSB as either the primary usb stack or selectable via menu Tools->USB Stack. You only need to include <Adafruit_TinyUSB.h> in your sketch to use.
> 
> adafruit/Adafruit_nRF52_Arduino
> 
> adafruit/ArduinoCore-samd
> 
> earlephilhower/arduino-pico
> 
> espressif/arduino-esp32
> 
> ESP32 port relies on Espressif's esp32-hal-tinyusb.c for building usb descriptors which requires all descriptors must be specified in usb objects declaration i.e constructors. Therefore all descriptor-related fields must be part of object declaration and descriptor-related API have no effect afterwards for this port.

https://docs.tinyusb.org/en/latest/info/uses.html

**Comment by fpistm** [comment, marked as answer]
No
